src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,16 +1,62 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Arrays;
 
 public class StorageImpl<K, V> implements Storage<K,V> {
+    private final int maxSize = 16;
+    private Object[] keys;
+    private Object[] values;
+    private double loadFactory = 0.75;
+    private int curCapacity;
+
+    public StorageImpl() {
+        keys = new Object[maxSize];
+        values = new Object[maxSize];
+        curCapacity = 1;
+        keys[0] = null;
+    }
+
+    private void extCapacity() {
+        if (curCapacity >= loadFactory * maxSize) {
+            values = Arrays.copyOf(values, values.length + values.length >> 1);
+            keys = Arrays.copyOf(keys, keys.length + keys.length >> 1);
+        }
+    }
+
+    private int isKeyThere(K key) {
```

Comment:
```suggestion
    private boolean isKeyExist(K key) {
```
Your method int isKeyThere is not required

Reply:
or you may call it indexOfElement